### PR TITLE
Verlet in X86_64

### PR DESCRIPTION
**Pull request by Gathros**

### CODE REVIEW

File: contents/verlet_integration/code/asm-x64/verlet.s [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  addsd  xmm3, xmm2                  # Adding dt to time
+  movsd  xmm4, xmm0                  # Hold old value of posistion
+  addsd  xmm0, xmm0                  # Calculating new position
+  subsd  xmm0, xmm5
+  addsd  xmm0, xmm6
+  movsd  xmm5, xmm4
+  comisd xmm0, xmm7                  # Check if position is greater then 0.0
+  ja     verlet_loop
+verlet_return:
+  movsd  QWORD PTR [rdi], xmm3       # Saving time value
```
Comment (Gustorn) [inline contents/verlet_integration/code/asm-x64/verlet.s]: You should just return in `xmm0` and `xmm1` (if necessary), passing in the pointer to return floats is a bit weird

File: contents/verlet_integration/code/asm-x64/verlet.s [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  addsd  xmm3, xmm2                  # Adding dt to time
+  movsd  xmm4, xmm0                  # Hold old value of posistion
+  addsd  xmm0, xmm0                  # Calculating new position
+  subsd  xmm0, xmm5
+  addsd  xmm0, xmm6
+  movsd  xmm5, xmm4
+  comisd xmm0, xmm7                  # Check if position is greater then 0.0
+  ja     stormer_verlet_loop
+stormer_verlet_return:
+  movsd  QWORD PTR [rdi], xmm3       # Saving time and velocity
```
Comment (Gustorn) [inline contents/verlet_integration/code/asm-x64/verlet.s]: Similarly, return in `xmm0` and `xmm1`

File: contents/verlet_integration/code/asm-x64/verlet.s [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  mulsd  xmm6, xmm2                  # Calculate new position
+  addsd  xmm6, xmm5
+  addsd  xmm0, xmm6
+  addsd  xmm4, xmm2                  # Incrementing time
+  movsd  xmm3, xmm4                  # Updating velocity
+  mulsd  xmm3, xmm1
+  comisd xmm0, xmm7
+  ja     velocity_verlet_loop
+velocity_verlet_return:
+  movsd  QWORD PTR [rdi], xmm4       # Saving time and velocity
```
Comment (Gustorn) [inline contents/verlet_integration/code/asm-x64/verlet.s]: Similarly, return in xmm0 and xmm1

**Comment by Liikt** [comment]
I would agree with @Gustorn here

**Pull request merged**